stackrox/stackrox
# ROX-9919 Make Central provide additional alert info about related network policies

## Description

This PR enables `alerts` API to return additional information about NetworkPolicies that apply to the Deployment.

If a deployment has ingress policy, but misses egress policies, the following object is returned:

![Screenshot 2022-04-08 at 11 58 03](https://user-images.githubusercontent.com/114479/162413204-326af9f8-579d-475c-9874-4236089f7ff7.png)

If a deployment misses both policies, then the following object is returned:

![Screenshot 2022-04-08 at 13 57 30](https://user-images.githubusercontent.com/114479/162431072-44ad5fff-5922-4cb8-acde-969bbf6c2473.png)

👉  @alwayshooin @pedrottimark please let me know if the new info returned by the API (and the new violation type) would be enough for you for the work planned it ROX-9760.

----

Note that the UI does not consume these objects yet. The UI view looks as follows:

![Screenshot 2022-04-08 at 11 58 12](https://user-images.githubusercontent.com/114479/162413291-e2ad99bf-4250-4f51-8503-3ebd2ca8a0ac.png)

Preview of the UI information (draft - the Frontend code is not part of this PR):

![Screenshot 2022-04-07 at 18 00 41](https://user-images.githubusercontent.com/114479/162430474-89a68261-a072-471f-a4ca-ad26b64d1762.png)
Implementing a view similar to this one should be the goal of ROX-9760.

## Checklist

## Testing Performed

- [x] Manually tested. Test steps:
  - Deploy local cluster enabling the feature-flag 
    - `export ROX_NETPOL_FIELDS=true`, or
    - `kubectl -n stackrox set env deploy/sensor ROX_NETPOL_FIELDS=true` and `kubectl -n stackrox set env deploy/central ROX_NETPOL_FIELDS=true`
  - Insert policies using `roxcurl`
    - `roxcurl /v1/policies -H "Content-Type: application/json" -i -X POST --data @api-policy-eg.json` 
    -  `roxcurl /v1/policies -H "Content-Type: application/json" -i -X POST --data @api-policy-ing.json`
  - Navigate to violations and observe deployments missing network policies


### Policies used in manual tests:

#### ` api-policy-eg.json`

```json
{
  "name": "Deployments should have egress Network Policy",
  "severity": "MEDIUM_SEVERITY",
  "disabled": false,
  "lifecycleStages": [
    "DEPLOY"
  ],
  "eventSource": "NOT_APPLICABLE",
  "isDefault": false,
  "categories": [
    "Anomalous Activity"
  ],
  "policySections": [
    {
      "sectionName": "No missing policies",
      "policyGroups": [
        {
          "fieldName": "Missing Egress Network Policy",
          "values": [
            {
              "value": "true"
            }
          ]
        }
      ]
    }
  ]
}
```

#### ` api-policy-ing.json`

```json
{
  "name": "Deployments should have ingress Network Policy",
  "severity": "MEDIUM_SEVERITY",
  "disabled": false,
  "lifecycleStages": [
    "DEPLOY"
  ],
  "eventSource": "NOT_APPLICABLE",
  "isDefault": false,
  "categories": [
    "Anomalous Activity"
  ],
  "policySections": [
    {
      "sectionName": "No missing policies",
      "policyGroups": [
        {
          "fieldName": "Missing Ingress Network Policy",
          "values": [
            {
              "value": "true"
            }
          ]
        }
      ]
    }
  ]
}
```
